Loop/Managers/NightscoutDataManager.swift

```diff
@@ -0,0 +1,113 @@
+            }
+        }
+
+        
+    }
+    
+    private var lastTempBasalUploaded: DoseEntry?
+
+    private func uploadLoopStatus(insulinOnBoard: InsulinValue?, predictedGlucose: [GlucoseValue]?, recommendedTempBasal: LoopDataManager.TempBasalRecommendation?, recommendedBolus: Double?, lastTempBasal: DoseEntry?, lastLoopError: ErrorType?) {
+        
+        let statusTime = NSDate()
+        
+        // Is this just predictedGlucose[0]?
+        let glucose = deviceDataManager.glucoseStore?.latestGlucose
+        
+        let iob: IOBStatus?
+        
+        if let insulinOnBoard = insulinOnBoard {
+            iob = IOBStatus(timestamp: insulinOnBoard.startDate, iob: insulinOnBoard.value)
+        } else {
+            iob = nil
+        }
+        
+        let eventualBG = predictedGlucose?.last?.quantity
+        
+        let loopSuggested: LoopSuggested?
+        
+        let glucoseVal = glucose?.quantity
+        
+        let predBGs: PredictedBG?
+        
+        if let predicted = predictedGlucose {
+            let values = predicted.map({ (value) -> HKQuantity in
+                value.quantity
+            })
+            predBGs = PredictedBG(values: values)
+        } else {
+            predBGs = nil
+        }
+        
+        if let recommendation = recommendedTempBasal, let glucoseVal = glucoseVal, let eventualBG = eventualBG {
+            loopSuggested = LoopSuggested(timestamp: recommendation.recommendedDate, rate: recommendation.rate, duration: recommendation.duration, eventualBG: eventualBG, bg: glucoseVal, correction: recommendedBolus, predBGs: predBGs)
+        } else {
+            loopSuggested = nil
+        }
+        
+        let loopEnacted: LoopEnacted?
+        if let tempBasal = lastTempBasal where tempBasal.unit == .unitsPerHour &&
+            lastTempBasalUploaded?.startDate != tempBasal.startDate {
+            let duration = tempBasal.endDate.timeIntervalSinceDate(tempBasal.startDate)
+            loopEnacted = LoopEnacted(rate: tempBasal.value, duration: duration, timestamp: tempBasal.startDate, received:
+                true)
+            lastTempBasalUploaded = tempBasal
+        } else if let recommendation = recommendedTempBasal {
+            // notEnacted
+            loopEnacted = LoopEnacted(rate: recommendation.rate, duration: recommendation.duration, timestamp: recommendation.recommendedDate, received: false)
+        } else {
+            loopEnacted = nil
+        }
+        
+        let loopName = NSBundle.mainBundle().bundleDisplayName
+        let loopVersion = NSBundle.mainBundle().shortVersionString
+        
+        let loopStatus = LoopStatus(name: loopName, version: loopVersion, timestamp: statusTime, iob: iob, suggested: loopSuggested, enacted: loopEnacted, failureReason: lastLoopError)
+        
+        deviceDataManager.remoteDataManager.uploadDeviceStatus(nil, loopStatus: loopStatus)
```

Comment:
Maybe this implementation can move into here, now, to keep all the NS models together?

Also, would be good to punt at the top of this method if there's no nightscout uploader